moduleframework/mtf_scheduler.py

```diff
@@ -0,0 +1,163 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+import argparse
+
+import os
+import logging
+import sys
+import yaml
+import moduleframework
+import subprocess
+import tempfile
+import shutil
+
+from sys import argv
+
+from avocado.utils import process
+from avocado.core.app import AvocadoApp
+from avocado.core.parser import Parser
+
+logger = logging.getLogger("mtf")
+
+
+def set_logging(level=logging.INFO):
+    global logger
+    logger.setLevel(level)
+
+    handler = logging.StreamHandler(sys.stderr)
+    handler.setLevel(logging.DEBUG)
+
+    formatter = logging.Formatter(
+        '%(asctime)s %(levelname)-6s %(message)s', '%H:%M:%S')
+    handler.setFormatter(formatter)
+    logger.addHandler(handler)
+
+    mekk_logger = logging.getLogger("mekk.xmind")
+    null_handler = logging.NullHandler()
+    mekk_logger.addHandler(null_handler)
+
+def cli():
+    # check needed/compulsory options
+    try:
+        someVariable = str(os.environ["MODULE"])
+        logger.debug("MODULE={MODULE}".format(MODULE=someVariable))
+    except KeyError:
+        logger.debug("Oops!, no MODULE")
+
+    parser = argparse.ArgumentParser(
+        # TODO
+        description="""Start your tests!, unknown param are forwarded to avocado \n  USE param like abc,tag TODO par """, formatter_class=argparse.RawTextHelpFormatter
+    )
+    parser.add_argument('--verbose', '-v', action='store_true', default=False)
+    parser.add_argument("--name", "-n", action="store", default="name not given", help='Name of module for testing')
+    parser.add_argument("--linter", "-l", action="store_true", default=False, help='Name of module for testing')
+    parser.add_argument("--setup", "-s", action="store_true", default=False, help='Setup')
+    parser.add_argument("--action", "-a", action="store", default='run', help='Action for avocado')
+    parser.add_argument("--module", "-m", action="store", default='docker', help='Module type, like: docker, nspawn or rpm')
+
+    #parser.add_argument('files', type=argparse.FileType('r'), nargs='+')
+
+    # load args from cli and show loaded options
+    args, unknown = parser.parse_known_args()
+
+    set_logging(level=logging.DEBUG if args.verbose else logging.INFO)
+
+    logger.debug("Options: verbose={0}, name={1}, linter={2}, setup={3}, action={4}, module={5}".format(
+        args.verbose, args.name, args.linter, args.setup, args.action, args.module))
+    logger.debug("remaining options for avocado: {0}".format(unknown))
+
+    # environment
+    # hint for read: os.environ.get('MODULE') ; write os.environ['MODULE']
+    if os.environ.get('MODULE') is None:
```

Comment:
I guess, if MODULE is not specified, than it should failed. It is important, isn't it?

Reply:
NOW  MODULE could be from:
    #   1. common.get_module_type() ... it reads config.yaml and treceback if it doesn't exist
    #   2. environment ... MODULE=docker etc
    #   3. argument ... --module=docker